### PR TITLE
feat: add cursorPaginate() support

### LINKED ISSUE #1197: `cursorPaginate()` doesn't work with `through()`
- Larastan Version: 2.1.2
- `--level` used: 5 (occurs from level 2)
- Pull request with failing test:

### Description
When using `->cursorPaginate()->through()` larastan reports: `Call to an undefined method Illuminate\Contracts\Pagination\CursorPaginator::through()`, however when using `->paginate()->through()` larastan doesn't

### Laravel code where the issue was found

Works:
```php
return User::paginate()
            ->through(function(User $user) {
                return [
                    'name' => Str::upper($user->name),
                    'email' => $user->email,
                ];
            });
```

Doesn't work:
```php
return User::cursorPaginate()
            ->through(function(User $user) {
                return [
                    'name' => Str::upper($user->name),
                    'email' => $user->email,
                ];
            });
```

Same when using DB:
```php
return DB::table('users')
            ->select([
                'name',
                'email',
            ])
            ->orderBy('id')
            ->paginate()
            ->through(function($user): array {
                return [
                    'name' => Str::upper($user->name),
                    'email' => $user->email,
                ];
            });
```
and
```php
return DB::table('users')
            ->select([
                'name',
                'email',
            ])
            ->orderBy('id')
            ->cursorPaginate()
            ->through(function($user): array {
                return [
                    'name' => Str::upper($user->name),
                    'email' => $user->email,
                ];
            });
```

### PR DESCRIPTION
**Pull request by BramVanBerkel**
- [x] Added or updated tests
- [x] Documented user facing changes (none?)
- [x] Updated CHANGELOG.md

Fixes #1197

**Changes**

Adds supoprt for the `cursorPaginate()` paginator

**Breaking changes**

None

### CODE REVIEW

**Comment by BramVanBerkel** [comment]
This fixes https://github.com/nunomaduro/larastan/issues/1197

**Comment by szepeviktor** [comment]
Thank you for your work, Bram!

**Comment by BramVanBerkel** [comment]
@szepeviktor I think this fixed it, however the unit tests are still failing due to 1 failure:
```
1) Type\GeneralTypeTest::testFileAsserts with data set "/Users/.../larastan/tests/Type/data/request-object.php:18" ('type', '/Users/.../Document...ct.php', PHPStan\Type\Constant\ConstantStringType Object (...), PHPStan\Type\UnionType Object (...), 18)
Expected type Illuminate\Foundation\Auth\User|null, got type App\User|null in /Users/.../larastan/tests/Type/data/request-object.php on line 18.
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'Illuminate\Foundation\Auth\User|null'
+'App\User|null'
```

I have tracked it down to this commit: https://github.com/nunomaduro/larastan/commit/06e932b689c63b2e4780e0aaa6143f913896f574

How should we approach this?

**Review by szepeviktor — APPROVED** [review]

**Comment by canvural** [comment]
Hi,

Can you rebase this on master? The changes in CHANGELOG does not look right. Otherwise it looks fine 👍🏽

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 * fix: updated `composer/pcre` to new PHP relevant version
 * feat: add support for the `foreignId` column in migrations
+* fix: add support for the `cursorPaginate()` paginator
```
Comment (szepeviktor) [inline CHANGELOG.md:11]:
```suggestion
* feat: add support for the `cursorPaginate()` paginator
```

**Comment by BramVanBerkel** [comment]
Updated the changelog, thanks both for your help and for your work on this package!

**Comment by canvural** [comment]
> Updated the changelog, thanks both for your help and for your work on this package!

You did not rebase it though. Anyway I fixed it in https://github.com/nunomaduro/larastan/commit/17d72fdeb3d4f6ecfb3cc043daac8df766c2c77a

Thank you!

**Comment by szepeviktor** [comment]
I've pressed a button here on GitHub.

kép

Isn't that rebasing?

**Pull request merged**